### PR TITLE
translate "search all cases" string

### PR DESCRIPTION
**Pull request by jmtroth0**
I don't know much about translations, but it seems like this should fix this? https://manage.dimagi.com/default.asp?282024

@kaapstorm @proteusvacuum buddy: @emord

### CODE REVIEW

File: corehq/apps/app_manager/models.py [hunk 1]
```diff
@@ -2392,7 +2392,7 @@ class CaseSearch(DocumentSchema):
     """
     Properties and search command label
     """
-    command_label = DictProperty(default={'en': 'Search All Cases'})
+    command_label = DictProperty(default=ugettext_lazy('Search All Cases'))
```
Comment (emord) [inline corehq/apps/app_manager/models.py]: This is setting the default to a string and not a dictionary. though farid can give more context of how this is used

**Comment by jmtroth0** [comment]
@proteusvacuum oh that's how that works. I found that line, it looks like `trans` will always take the default since there isn't anything else setting the "command_label" property. I figured it would be fine if the default was in the language of the user.

It seems like what we want instead then would be to provide the option of languages to send down to the phone? Or just to send all of them every time? Either way, it seems like this is a feature request rather than an interrupt bug

**Comment by proteusvacuum** [comment]
@jmtroth0 we already send all the languages to the phone. 

Take a look at the app strings here: https://www.commcarehq.org/a/sci-civ-malaria/apps/download/adc71f6a7c7d81c36c89636adf23eab6/

The problem is that the correct keys aren't getting added to the bulk translations file or being auto filled in the UI. That is what the bug is here.

The workaround solution is for this domain is to add a custom translation key of the form "case_search.m{x}" with the translation they want. 

Also adding the above in the ticket.

**Comment by jmtroth0** [comment]
@proteusvacuum I get that. That doesn't seem to solve the issue generally though. It doesn't look like we generate alternatives at all for that particular string. For all domains it will only ever be the default which is hardcoded in, because of how the `trans` function works, which is why even in the french translation it is being set as "search all cases". 

Do we have another translation scaffolding for places like this where we need android style translations on the python side?

**Comment by proteusvacuum** [comment]
It doesn't look like we provide any default translations for the CommCare UI. If you bulk download the translations file for an application that has french as a language, none of it is populated. So I think the answer to that, unfortunately, is no.

**Comment by orangejenny** [comment]
The ticket has an image of the domain-level case config page, but I'd expect this string could vary by app or even by module (as the `m{x}` in `case_search.m{x}` implies), and that there should really be an input like "Button label" somewhere in the case list section:
screen shot 2018-09-07 at 12 33 26 am

@proteusvacuum does that sound like the right ideal state?

If it is, I think making the UI support translations properly is a feature request (localizing the "Display Text" in that UI seems broken...I'm guessing this feature never really supported translations), but making bulk app translations support the command label shouldn't be hard. @jmtroth0 if you want to take a crack at that, look at how `case_list_form_label` is handled in [app_translations](https://github.com/dimagi/commcare-hq/tree/master/corehq/apps/app_manager/app_translations/app_translations.py). If you're not in the mood, kick the ticket over to me, since it's app manager.

**Comment by proteusvacuum** [comment]
@orangejenny that option sounds right to me. 
What makes you think the display text localization doesn't work? It works for me when I switch the language, and the correct keys show up in the app_strings.

**Comment by orangejenny** [comment]
@proteusvacuum I added display text [here](https://www.commcarehq.org/a/jennytraining/apps/view/c8098a1aa82f4105e4b69e861b149d23/module/b8df7e2d265f07f1f83c18465dd06990f9178a29/?lang=fra#case-detail-screen-config-tab) while in french, saved, switched to english, switched back, and it was gone.

**Pull request closed without merging**